packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -504,6 +499,209 @@ void promiseArray;
 ['I', 'am', 'just', 'an', 'array'];
       `,
     },
+    // type annotations on variables containing promises
```

Comment:
Nit: other rule tests -both in this file and mostly in general for us- don't do this

```suggestion
```

IMO when there are a whole bunch of rule tests like this, it should be apparent what each one / group is testing. They tend to subtly change over time.

...that being said, I am finding some of the more nuanced cases below to be useful. So 🤷 for this I don't feel strongly where the line between "do" and "don't" is.

Reply:
trimmed em down a bit 👍